src/UCommerce.Transactions.Payments.PayPal/PayPalWebSitePaymentsStandardPageBuilder.cs

```diff
@@ -6,93 +6,117 @@
 using Ucommerce.Extensions;
 using Ucommerce.Transactions.Payments.Configuration;
 using Ucommerce.Transactions.Payments.Common;
+using Ucommerce.Web;
 
 namespace Ucommerce.Transactions.Payments.PayPal
 {
-	public class PayPalWebSitePaymentsStandardPageBuilder : AbstractPageBuilder
-	{
-		private const string URL_SANDBOX = "https://www.sandbox.paypal.com/cgi-bin/webscr";
-		private const string URL_PRODUCTION = "https://www.paypal.com/cgi-bin/webscr";
+    public class PayPalWebSitePaymentsStandardPageBuilder : AbstractPageBuilder
+    {
+        private readonly IAbsoluteUrlService _absoluteUrlService;
+        private readonly ICallbackUrl _callbackUrl;
 
-		public string GetPostUrl(PaymentMethod paymentMethod)
-		{
-			return paymentMethod.DynamicProperty<bool>().Sandbox
-				? URL_SANDBOX
-				: URL_PRODUCTION;
-		}
+        private const string URL_SANDBOX = "https://www.sandbox.paypal.com/cgi-bin/webscr";
+        private const string URL_PRODUCTION = "https://www.paypal.com/cgi-bin/webscr";
 
-		public PayPalWebSitePaymentsStandardPageBuilder() { }
+        public string GetPostUrl(PaymentMethod paymentMethod)
+        {
+            return paymentMethod.DynamicProperty<bool>().Sandbox
+                ? URL_SANDBOX
+                : URL_PRODUCTION;
+        }
 
-		protected override void BuildHead(StringBuilder page, PaymentRequest paymentRequest)
-		{
-			page.Append("<title>Paypal</title>");
-			page.Append(@"<script type=""text/javascript"" src=""//ajax.googleapis.com/ajax/libs/jquery/1.4.2/jquery.min.js""></script>");
-			if (!Debug)
-				page.Append(@"<script type=""text/javascript"">$(function(){ $('form').submit();});</script>");
-		}
+        public PayPalWebSitePaymentsStandardPageBuilder(IAbsoluteUrlService absoluteUrlService,
+            ICallbackUrl callbackUrl)
+        {
+            _absoluteUrlService = absoluteUrlService;
+            _callbackUrl = callbackUrl;
+        }

```

Comment:
Mandatory property injection instead? 😊 It would already stop us from breaking PayPalWebSitePaymentsStandardRecurringPaymentPageBuilder as well :P It can also make customers happier

Reply:
Done! Thanks